torchdiffeq/_impl/misc.py

```diff
@@ -228,31 +241,35 @@ def _check_inputs(func, y0, t, rtol, atol, method, options, event_fn, SOLVERS):
             options['norm'] = _mixed_linf_rms_norm(shapes)
 
     # Normalise time
-    assert torch.is_tensor(t), 't must be a torch.Tensor'
-    _assert_one_dimensional('t', t)
-    _assert_floating('t', t)
-    decreasing_time = False
-    if _decreasing(t):
-        decreasing_time = True
+    t = _check_timelike('t', t, True)
+    t_is_reversed = False
+    if len(t) > 1 and t[0] > t[1]:
+        t_is_reversed = True
+
+    if t_is_reversed:
+        # Change the integration times to ascending order.
+        # We do this by negating the time values and all associated arguments.
         t = -t
+
+        # Ensure time values are un-negated when calling functions.
         func = _ReverseFunc(func, mul=-1.0)
         if event_fn is not None:
             event_fn = _ReverseFunc(event_fn)
+
+        # For fixed step solvers.
         try:
-            grid_points = options['grid_points']
+            _grid_constructor = options['grid_constructor']
         except KeyError:
             pass
         else:
-            options['grid_points'] = -grid_points
+            options['grid_constructor'] = lambda func, y0, t: -_grid_constructor(func, y0, -t)
+
+        # For RK solvers.
+        _flip_option(options, 'step_t')
+        _flip_option(options, 'jump_t')
```

Comment:
This is unnecessary if these are assumed to be unordered? (And I think `_flip_option` removed.)

Reply:
This one negates the values and is necessary when values of t are negated i think.